### ISSUE TITLE
exif_transpose(img) TypeError exception on file that GIF and Javascript are happy to process

### ISSUE THREAD

**Issue by johngrabner** [python-pillow/Pillow#7340]
### What did you do?
```python
        try:
            img = ImageOps.exif_transpose(img)
        except TypeError:
            print(f"TypeError occurred!, file {img_path}")
```

### What did you expect to happen?
rotate the file 90 degrees

### What actually happened?
exception TypeError

error occurs here
```
def tobytes(self, offset=8):
   ...
   return b"Exif\x00\x00" + head + ifd.tobytes(offset) 

TiffImagePlugin.py:: tobytes(self, offset=0):
   ...
   for tag, value in sorted(self._tags_v2.items()): ..... 41th interation tag ==5, value=0
       ...
       logger.debug(f"Tag {tag}, Type: {typ}, Value: {repr(value)}")
       'Tag 5, Type: 1, Value: 0'
       ...
       values = value if isinstance(value, tuple) else (value,)  .... value==(0,)
       data = self._write_dispatch[typ](self, *values) .... data==0
       ... 
      msg = f"save: {tagname} ({tag}) - type: {typname} ({typ})"
      .... at this point msg == 'save: GPSAltitudeRef (5) - type: byte (1)'
      ... this statement crashes because len(data) TypeError: object of type 'int' has no len()
      msg += " - value: " + (
      "<table: %d bytes>" % len(data) if len(data) >= 16 else str(values)
      )
```
      
debug logging causes the crash.

### What are your OS, Python and Pillow versions?

* OS:  Ubuntu 18.04.6 LTS (x86_64)
*   docker image  FROM pytorch/pytorch:1.13.1-cuda11.6-cudnn8-runtime
* Python: Python version: 3.10.8 (main, Nov  4 2022, 13:48:29) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.15.0-78-generic-x86_64-with-glibc2.27

* Pillow: '9.3.0'

Here is one of may files having this problem.

![20220721_181853](https://github.com/python-pillow/Pillow/assets/8209285/9ccc72ef-b7f5-4c87-bae7-1c5f33047778)

**Comment by radarhere** [comment]
Hi. Testing your image, this was fixed by #6740 in Pillow 9.4.0.

So the solution here is simply to upgrade your Pillow to a more recent version.

**Comment by johngrabner** [comment]
Thank you
